src/solvers/flattening/boolbv_add_sub.cpp

```diff
@@ -100,7 +112,8 @@ bvt boolbvt::convert_add_sub(const exprt &expr)
 
         for(std::size_t j=0; j<tmp_result.size(); j++)
         {
-          assert(i*sub_width+j<bv.size());
+          INVARIANT(
+            i * sub_width + j < bv.size(), "bit index shall be within bounds");
```

Comment:
As above: check once, before entering the loop. And then consider whether it can be checked even earlier (as soon as all variables' values are known).